hazelcast/src/main/java/com/hazelcast/jet/impl/connector/WriteJdbcP.java

```diff
@@ -130,19 +132,25 @@ public static <T> ProcessorMetaSupplier metaSupplier(
         return ProcessorMetaSupplier.preferLocalParallelismOne(
                 ConnectorPermission.jdbc(jdbcUrl, ACTION_WRITE),
                 new ProcessorSupplier() {
-                    private transient CommonDataSource dataSource;
+                    private transient DataStoreHolder<? extends CommonDataSource> dataSource;
 
                     @Override
                     public void init(@Nonnull Context context) {
                         dataSource = dataSourceSupplier.apply(context);
                     }
 
-                    @Nonnull @Override
+                    @Override
+                    public void close(Throwable error) throws Exception {
+                        dataSource.close();
```

Comment:
seems we don't need this newly introduced `close` method anymore since data-source is always `nonClosing`

Reply:
here we can pass a non-shared datastore which needs to be closed: https://github.com/hazelcast/hazelcast/blob/e3e5f468fa53e660297625dba4a7855908339b29/hazelcast/src/main/java/com/hazelcast/jet/core/processor/SinkProcessors.java#L421